src/Symfony/Component/Mailer/Bridge/Mailchimp/Transport/MandrillApiTransport.php

```diff
@@ -93,6 +93,10 @@ private function getPayload(Email $email, Envelope $envelope): array
             ],
         ];
 
+        if ($email->getHeaders()->get('X-MC-Subaccount')) {
```

Comment:
this header should probably be added in the list of headers that are skipped in the loop below, due to being handled in a special way.

Reply:
and the test should assert also that `$payload['message']['headers']` does not contain the header anymore.